### PR TITLE
pgdir and service can be absolute paths now.

### PR DESCRIPTION
**Pull request by bukzor**
@kentwills @adityavohra7 @asottile

### CODE REVIEW

File: pgctl/service.py [hunk 1]
```diff
@@ -117,13 +119,18 @@ def assert_ready(self):
...
         if not self.path.check(dir=True):
-            raise NoSuchService("No such playground service: '%s'" % self.name)
+            raise NoSuchService("No such service: '%s'" % bestrelpath(str(self.path)))
 
         # ensure symlink {service_dir}/supervise -> {scratch_dir}/supervise
+        # this will re-connect the service to its state descriptors if the symlinks have been deleted or moved
         supervise_in_scratch = self.scratch_dir.join('supervise')
         supervise_in_scratch.ensure_dir()
         symlink_if_necessary(supervise_in_scratch, self.path.join('supervise'))
+        self.__exists = True
```
Comment (asottile) [inline pgctl/service.py]: mutating a previously immutable object? :(

File: pgctl/cli.py [hunk 2]
```diff
@@ -19,19 +19,13 @@
...
 from .errors import Unsupervised
-from .flock import flock
-from .flock import set_fd_inheritable
 from .functions import commafy
 from .functions import exec_
 from .functions import JSONEncoder
-from .functions import ps
 from .functions import uniq
-from .fuser import fuser
 from .service import Service
```
Comment (asottile) [inline pgctl/cli.py]: unnecessary diff churn

this-style--

File: pgctl/service.py (hunk 1 above)
Comment (bukzor) [inline pgctl/service.py]: Not different from the `@cached_property`s.
Write-once =~ immutable
